### PR TITLE
Devspace docker issues

### PR DESCRIPTION
**Pull request by aleksandra-tarkowska**
This PR resolve https://trello.com/c/bfzIv5Pj/2428-ci-master-jenkins-upgrade :

- additional docker issues,
- request mandatory jenkins password (proxy basic http authentication password is optional and can be set only in vars)

### CODE REVIEW

File: ansible/requirements.yml [hunk 1]
```diff
@@ -44,8 +44,7 @@
   src: openmicroscopy.logrotate
   version: 1.0.0
 
-- name: lvm-partition
```
Comment (aleksandra-tarkowska) [inline ansible/requirements.yml]: @manics are you ok with that?

Comment (manics) [inline ansible/requirements.yml]: That'll break other playbooks that currently rely on `lvm-partition`.

**Comment by aleksandra-tarkowska** [comment]
I removed ccd961e

**Comment by joshmoore** [comment]
@manics : can you sign off on this?

@aleksandra-tarkowska : what version do you propose here? The variable name changes are nearly breaking, right? Have you tested this with Jenkins v1 as well as v2, or does it require v2?

**Comment by aleksandra-tarkowska** [comment]
How do you mean var names are changing? If someone had set up `devspace_jenkins_nginx_password` in vars it should remain and nginx basic auth will be still working. For those who set password manually it is not possible now to press Enter! (cc: @sbesson  ;-) )

I think devspace has to be treated as a framework. It is up to the maintainer what kind of authentication plugin would like to use (e.g. ldap, etc).

Note that I wasn't successful with credentials for nodes https://trello.com/c/eKkIBg0r/1854-add-an-admin-password-for-devspace-jenkins#comment-588cdf1327142b86bbc8a2c0.

**Comment by aleksandra-tarkowska** [comment]
the cahnge here is not about Jenkins v1 v2, is about compose https://github.com/openmicroscopy/devspace/pull/65/files#diff-4e5e90c6228fd48698d074241c2ba760R7  I think it should be transparent

**Comment by sbesson** [comment]
I am completely fine with storing a non-default password in a devspace inventory file. I had overlooked the fact this was already possible.

**Comment by aleksandra-tarkowska** [comment]
I would like to still discuss if realm strategy is good and will resolve all the issues cc @sbesson @joshmoore @manics @kennethgillen

File: ansible/roles/devspace/tasks/devspace-install.yml [hunk 2]
```diff
@@ -135,7 +137,7 @@
     line: "        auth_basic \"Restricted\";"
     insertafter: "    location / {"
     state: present
-  when: devspace_jenkins_nginx_username|length>0 and devspace_jenkins_nginx_password|length>0
+  when: devspace_jenkins_nginx_password and devspace_jenkins_nginx_username|length>0 and devspace_jenkins_nginx_password|length>0
```
Comment (manics) [inline ansible/roles/devspace/tasks/devspace-install.yml]: `devspace_jenkins_nginx_password` is defined in `defaults/main.yml` file, so you could simplfy this check by setting the default to `devspace_jenkins_nginx_password: ''` so you can just check the length here

**Comment by joshmoore** [comment]
Summary:

 * old variable is still supported, but now it can also be set via variable.
 * cF. https://github.com/aleksandra-tarkowska/ansible-role-devspace
 * @sbesson will test against the old devspace (re-deploying)

**Comment by aleksandra-tarkowska** [comment]
```
$ ansible-galaxy install aleksandra-tarkowska.devspace
- downloading role 'devspace', owned by aleksandra-tarkowska
- downloading role from https://github.com/aleksandra-tarkowska/ansible-role-devspace/archive/master.tar.gz
- extracting aleksandra-tarkowska.devspace to /Users/ola/OMERO/SYSADMIN/management_tools/infrastructure/ansible/vendor/aleksandra-tarkowska.devspace
- aleksandra-tarkowska.devspace was installed successfully
[DEPRECATION WARNING]: The comma separated role spec format, use the yaml/explicit format instead..
This 
feature will be removed in a future release. Deprecation warnings can be disabled by setting 
deprecation_warnings=False in ansible.cfg.
- adding dependency: openmicroscopy.upgrade-distpackages
- adding dependency: openmicroscopy.docker
- dependency openmicroscopy.versioncontrol-utils is already installed, skipping.
- downloading role 'upgrade-distpackages', owned by openmicroscopy
- downloading role from https://github.com/openmicroscopy/ansible-role-upgrade-distpackages/archive/master.tar.gz
- extracting openmicroscopy.upgrade-distpackages to /Users/ola/OMERO/SYSADMIN/management_tools/infrastructure/ansible/vendor/openmicroscopy.upgrade-distpackages
- openmicroscopy.upgrade-distpackages was installed successfully
- downloading role 'docker', owned by openmicroscopy
- downloading role from https://github.com/openmicroscopy/ansible-role-docker/archive/master.tar.gz
- extracting openmicroscopy.docker to /Users/ola/OMERO/SYSADMIN/management_tools/infrastructure/ansible/vendor/openmicroscopy.docker
- openmicroscopy.docker was installed successfully
- dependency openmicroscopy.lvm-partition is already installed, skipping.
```


and playbook

```
$ ansible-playbook -l devspace-test devspace.yml -u centos 
Secure Jenkins with admin password (username: devspace): 

PLAY [devspace] ****************************************************************
...
TASK [openmicroscopy.upgrade-distpackages : system packages | upgrade] *********
...
TASK [openmicroscopy.docker : upstream docker | setup repository] **************
...
TASK [aleksandra-tarkowska.devspace : add user omero to docker group] **********

```

**Comment by joshmoore** [comment]
> is there any doc somewhere indicating the ansible-galaxy step?

Happy to try to cover in https://github.com/openmicroscopy/infrastructure/pull/246

**Comment by joshmoore** [comment]
The requirements.yml changes needed for this PR are now in https://github.com/manics/infrastructure/pull/5

**Comment by aleksandra-tarkowska** [comment]
thx I will come back to it in next few days

**Comment by joshmoore** [comment]
Considering that this was largely waiting on requirements.yml fixes, this now looks to be done. The playbook isn't versioned so merging without a tag. Next changes could likely be done after a migration of the playbook to the role repository.

**Pull request merged**